### ISSUE TITLE
Neovim LanguageServer Usage

### ISSUE THREAD

**Issue by joshbode** [JuliaEditorSupport/LanguageServer.jl#100]
Not an issue (as in problem) but I have managed to get the Julia LanguageServer working with neovim :)

This is the minimal config I use to get it working (installed/managed via [vim-plug](https://github.com/junegunn/vim-plug)):

```
call g:plug#begin()
  Plug 'JuliaEditorSupport/julia-vim'
  Plug 'autozimu/LanguageClient-neovim', { 'do': ':UpdateRemotePlugins' }
  Plug 'roxma/nvim-completion-manager'
call g:plug#end()

" julia
let g:default_julia_version = '0.6'
let g:julia_syntax_highlight_deprecated = 1

" language server
let g:LanguageClient_autoStart = 1
let g:LanguageClient_serverCommands = {
\   'julia': ['julia', '--startup-file=no', '--history-file=no', '-e', '
\       using LanguageServer;
\       server = LanguageServer.LanguageServerInstance(STDIN, STDOUT, false);
\       server.runlinter = true;
\       run(server);
\   '],
\ }

nnoremap <silent> K :call LanguageClient_textDocument_hover()<CR>
nnoremap <silent> gd :call LanguageClient_textDocument_definition()<CR>
nnoremap <silent> <F2> :call LanguageClient_textDocument_rename()<CR>
```

Note: `JuliaEditorSupport/julia-vim` is required otherwise Neovim thinks `.jl` files are lisp and the client won't start. Plus, some syntax highlighting is always nice :)

Linting from the LanguageServer doesn't seem to work so I'm currently using NeoMake/Lint.jl for that (not shown).

**Comment by ZacLN** [comment]
Great! Would you mind making a PR to the readme with the config details?

**Comment by davidanthoff** [comment]
This is so cool!!!!

**Comment by joshbode** [comment]
I'd be happy to!
(n)vim + julia = 🎉

Is there supposed to be linting information coming from the Language Server currently? If so, I'm not seeing it currently under this config (the nvim language client supports linting for other languages, such as Python)

**Comment by joshbode** [comment]
Figured out the linting: need to set `runlinter = true` on the server object.

**Comment by ZacLN** [comment]
Fantastic!

Yep, diagnostic messages should be being transfered from server to client of the form: 
```
{"method":"textDocument/publishDiagnostics","params":{"uri":"untitled:Untitled-2","diagnostics":[{"range":{"start":{"line":0,"character":0},"end":{"line":0,"character":1}},"severity":2,"code":"PossibleTypo","source":"CSTParser.Diagnostics.Diagnostic","message":"Possible use of undeclared variable x"}]},"jsonrpc":"2.0"}
```

**Comment by joshbode** [comment]
Closed via #103 and wiki page
